src/components/AlertCard/index.tsx

```diff
@@ -1,78 +1,159 @@
+  variant: Variant
+  size?: Size
+  layout?: Layout
   isOpen?: boolean
   showClose?: boolean
-  icon?: JSX.Element
+  closeButton?: React.ReactNode
+  showIcon?: boolean
+  icon?: React.ReactNode
+  iconContainer?: boolean
+  listItems?: string[] | null
+  link?: {
+    text: string
+    onClick: () => void
+  }
+  buttons?: {
+    primary?: {
+      text: string
+      onClick: () => void
+    }
+    secondary?: {
+      text: string
+      onClick: () => void
+    }
+    tertiary?: {
+      text: string
+      onClick: () => void
+    }
+  }
+  noBorderLeft?: boolean
+  className?: string
 }
 
 export default function Alert({
   title,
   message,
-  actionText,
   onClose = () => {},
-  onActionClick = () => {},
   variant,
+  size = 'small',
+  layout = 'horizontal',
   isOpen = true,
   showClose = true,
+  closeButton = <CloseButton />,
+  showIcon = true,
   icon,
+  iconContainer = false,
+  listItems,
+  link,
+  buttons,
+  noBorderLeft = false,
+  className,
   ...props
 }: InfoAlertProps) {
+  if (!isOpen) return null
+
   return (
     <div
-      className={classNames(styles.container, styles[variant], {
-        [styles.closed]: !isOpen
-      })}
+      className={classNames(
+        styles.container,
+        styles[variant],
+        styles[size],
+        styles[layout],
+        {
+          [styles.noBorderLeft]: noBorderLeft,
+          [styles.iconContainer]: iconContainer
+        },
+        className
+      )}
       {...props}
     >
-      <div className={styles.icon}>{icon ?? <WarningIcon />}</div>
-      <div className={styles.text}>
-        <div className={classNames('button', styles.title)}>{title}</div>
-        <div
-          className={classNames(
-            {
-              'body-small': variant === 'warning',
-              body: variant === 'danger'
-            },
-            styles.message
-          )}
-        >
-          {message}
-        </div>
-      </div>
       {showClose ? (
-        <div className={styles.close}>
-          <CloseButton onClick={onClose} />
-        </div>
+        <button type="button" onClick={onClose} className={styles.closeButton}>
+          {closeButton}
+        </button>
       ) : null}
-      {actionText && (
-        <div className={styles.action}>
-          <Button
-            rightIcon={<RightArrow />}
-            onClick={onActionClick}
-            type="link"
-            size="small"
-          >
-            {actionText}
-          </Button>
+      {showIcon ? <div className={styles.icon}>{icon}</div> : null}
+      <div className={styles.content}>
+        <div className={styles.header}>
+          <div className={styles.title}>{title}</div>
         </div>
-      )}
+        <div className={styles.message}>{message}</div>
+
+        {listItems ? (
+          <ul className={styles.list}>
+            {listItems.map((item, index) => (
+              <li key={index}>{item}</li>
+            ))}
+          </ul>
+        ) : null}
```

Comment:
Read this comment after you read my two comments below.

I think this can also be passed it as the button group `React.ReactNode`